File: js/stockPicker.js
exports.picker = function (prices) {
  let buy_index = 0;
  let gains = 0;
  let sell_index = 0;

  for (let i = 1; i < prices.length; i++) {
    if (prices[i] < prices[buy_index]) {
      buy_index = i;
    } else if (prices[i] - prices[buy_index] > gains) {
      gains = prices[i] - prices[buy_index];
      sell_index = i;
    }
  }
  buy_index = prices.indexOf(Math.min(...prices.slice(0, sell_index)));
  return [buy_index, sell_index];
};

File: python/stock_picker.py
def picker(prices):
    buy_index = 0
    gains = 0
    sell_index = 0
    
    for i in range(1, len(prices)):
        if prices[i] < prices[buy_index]:
            buy_index = i
        elif prices[i] - prices[buy_index] > gains:
            gains = prices[i] - prices[buy_index]
            sell_index = i
    buy_index = prices.index(min(prices[0:sell_index]))
    return [buy_index, sell_index]